src/application.c

```diff
@@ -246,7 +246,7 @@ static void save_state_to_file_or_folder(struct swappy_state *state,
 }
 
 static void maybe_save_output_file(struct swappy_state *state) {
-  if (state->output_file != NULL) {
+  if (state->config->auto_save) {
```

Comment:
```suggestion
  if (state->config->auto_save && state->output_file != NULL) {
```

Reply:
I dont think we need to check NULL for state->output_file here, as in case output_file=NULL, in `save_state_to_file_or_folder` the file name is generated. 

Actually, in my case, with the NULL check, at least in my case, the file is not auto save.
I'm using swappy in `sway` using the command as specified in the README: 

```sh 
grim -g "$(slurp)" - | swappy -f -
```
every time I make a screenshot, then I close the swappy, the filename is NULL so the picture is not saved.